controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java

```diff
@@ -317,7 +323,16 @@ private void assignedRotationsToSlime(List<AssignedRotation> rotations, Cursor p
 
     // ------------------ Deserialization
 
-    public Application fromSlime(Slime slime) {
+    public Application fromSlime(byte[] data) {
+        var key = Arrays.hashCode(data);
+        try {
+            return cache.get(key, () -> fromSlime(SlimeUtils.jsonToSlime(data)));
```

Comment:
Sure, but I want to avoid keeping a reference to the byte array. Changed to one of the [hashing functions from Guava](https://github.com/google/guava/wiki/HashingExplained).

Reply:
I guess that's good enough :)

Reply:
From a cursory glance, it would seem the sip hash function is a safe, collision resistant function. Perhaps use that instead, as the current one comes with no guarantees.

Reply:
PTAL